### PR TITLE
Modify accumulo-cluster to support starting and stopping groups of tservers

### LINKED ISSUE #5109: accumulo-cluster needs to be modified to support tserver groups
Tserver groups are not an option in previous versions of accumulo. 
Starting in 4.0, the accumulo-cluster script should be modified to support starting tservers per resource group.

### PR DESCRIPTION
**Pull request by ddanielr**
Closes #5109 

Required modifications to ZooZap to support resource groups.

### CODE REVIEW

**Comment by ddanielr** [comment]
Tested locally and used `./accumulo dump-zoo` to verify zookeeper contents.

File: assemble/bin/accumulo-cluster [hunk 1]
```diff
@@ -32,9 +32,9 @@ Commands:
   stop                       Stops Accumulo cluster
   kill                       Kills Accumulo cluster
   start-non-tservers         Deprecated. Starts all services except tservers
```
Comment (dlmarion) [inline assemble/bin/accumulo-cluster:34]: I'm wondering if we should remove the deprecated commands at this point. Were they marked deprecated in an older release?

File: assemble/bin/accumulo-cluster [hunk 2]
```diff
@@ -32,9 +32,9 @@ Commands:
...
-  start-servers [--all|--tservers|--no-tservers|--sservers [group]|--compactors [group]]
+  start-servers [--all|--tservers [group]|--no-tservers|--sservers [group]|--compactors [group]]
```
Comment (dlmarion) [inline assemble/bin/accumulo-cluster:35]: I'm thinking we should either get rid of the `--no-tservers` option or add `--no-sservers` and `--no-compactors`.

File: assemble/bin/accumulo-cluster [hunk 3]
```diff
@@ -273,7 +273,12 @@ function stop_sservers() {
 
 function start_tservers() {
   echo "Starting tablet servers ..."
-  for group in $TSERVER_GROUPS; do
+  groups=$TSERVER_GROUPS
+  if [[ -n $1 ]]; then
+    groups="$1"
+    echo "Only stopping servers for group: ${groups}"
```
Comment (dlmarion) [inline assemble/bin/accumulo-cluster:279]:
```suggestion
    echo "Only starting servers for group: ${groups}"
```

**Review by dlmarion — COMMENTED** [review]
A change and some unrelated comments.

File: assemble/bin/accumulo-cluster (hunk 2 above)
Comment (dlmarion) [reply assemble/bin/accumulo-cluster:35]: I think I'm not familiar with the use case for starting everything but the tservers.

File: assemble/bin/accumulo-cluster (hunk 1 above)
Comment (ddanielr) [reply assemble/bin/accumulo-cluster:34]: Yes they were marked deprecated in 2.1. I was going to put up another PR removing them.

File: assemble/bin/accumulo-cluster (hunk 2 above)
Comment (ddanielr) [reply assemble/bin/accumulo-cluster:35]: That was a carry-over from the `start-non-tservers` behavior. I think it's probably best if we remove the `--no-tservers` option vs adding extra flags.

File: assemble/bin/accumulo-cluster [hunk 4]
```diff
@@ -439,7 +448,7 @@ function stop_tservers() {
 
   echo "Cleaning tablet server entries from zookeeper"
   if ! isDebug; then
-    ${accumulo_cmd} org.apache.accumulo.server.util.ZooZap -tservers
+    ${accumulo_cmd} org.apache.accumulo.server.util.ZooZap -tservers -group "$groups"
```
Comment (dlmarion) [inline assemble/bin/accumulo-cluster:451]: In the case where a group is not specified, then `$groups` is going to be all groups. Does ZooZap handle that correctly?

File: assemble/bin/accumulo-cluster (hunk 2 above)
Comment (ddanielr) [reply assemble/bin/accumulo-cluster]: Flags have been removed.

File: assemble/bin/accumulo-cluster (hunk 1 above)
Comment (ddanielr) [reply assemble/bin/accumulo-cluster]: Ended up just making the changes here to avoid merge conflicts.

**Review by dlmarion — APPROVED** [review]
LGTM. I think there is further simplification that could be done in another PR. `start` starts everything and `start-servers` either starts everything or starts only one type of server process. Seems like we could simplify this to:

`start [--all (default) | [--manager] [--gc] [--monitor] [--tservers [group]] [--sservers [group]] [--compactors [group]]`

File: assemble/bin/accumulo-cluster (hunk 4 above)
Comment (ddanielr) [reply assemble/bin/accumulo-cluster]: Good catch, moved that section up to the last for loop so it's handled correctly.

**Comment by dlmarion** [comment]
> LGTM. I think there is further simplification that could be done in another PR. `start` starts everything and `start-servers` either starts everything or starts only one type of server process. Seems like we could simplify this to:
> 
> `start [--all (default) | [--manager] [--gc] [--monitor] [--tservers [group]] [--sservers [group]] [--compactors [group]]`

Created #5116 for this. I haven't tested it yet, but I'm thinking it should work.

**Pull request merged**